lib/src/main/java/io/ably/lib/transport/Hosts.java

```diff
@@ -40,39 +40,38 @@
      */
     public Hosts(String primaryHost, String defaultHost, ClientOptions options) throws AblyException {
         this.defaultHost = defaultHost;
-        if (primaryHost != null) {
+        this.fallbackHostsUseDefault = options.fallbackHostsUseDefault;
+        boolean hasCustomPrimaryHost = primaryHost != null && !primaryHost.equalsIgnoreCase(defaultHost);
+        String[] tempFallbackHosts = options.fallbackHosts;
+        if (options.fallbackHostsUseDefault) {
+            if (options.fallbackHosts != null) {
+                throw AblyException.fromErrorInfo(new ErrorInfo("fallbackHosts and fallbackHostsUseDefault cannot both be set", 40000, 400));
+            }
+            if (options.port != 0 || options.tlsPort != 0) {
+                throw AblyException.fromErrorInfo(new ErrorInfo("fallbackHostsUseDefault cannot be set when port or tlsPort are set", 40000, 400));
+            }
+            tempFallbackHosts = Defaults.HOST_FALLBACKS;
+        }
+
+        boolean isProduction = options.environment == null || options.environment.isEmpty() || "production".equalsIgnoreCase(options.environment);
+
+        if (!hasCustomPrimaryHost && tempFallbackHosts == null && options.port == 0 && options.tlsPort == 0) {
+            tempFallbackHosts = isProduction ? Defaults.HOST_FALLBACKS : Defaults.getEnvironmentFallbackHosts(options.environment);
+        }
+
+        if (hasCustomPrimaryHost) {
```

Comment:
!hasCustomPrimaryHost

Reply:
Could you please elaborate on why it should be inverted?

Reply:
Wow, I didn't see line no 59. That explains the logic. It should work now : )
I will mark as resolved.